src/CoreWCF.Primitives/src/CoreWCF/Channels/Message.cs

```diff
@@ -748,7 +751,14 @@ public virtual async Task OnWriteMessageAsync(XmlDictionaryWriter writer)
             // otherwise EnsureWriteMessageState would get called twice. Also see OnWriteMessage()
             // for the example.
             await OnWriteBodyContentsAsync(writer);
-            WriteMessagePostamble(writer);
+            if (writer.SupportsAsync())
```

Comment:
You probably want to update BodyWriterMessage.OnWriteMessageAsync with the same logic too. That class/method is further down in this same file.